### PR TITLE
[docs] docs: update MoE dispatcher selection guidance

### PR DESCRIPTION
**Pull request by yaoyu-33**
## Summary
- Add dispatcher backend availability checks to the MoE optimization docs and dispatcher-selection skill.
- Record the 2026-05-17 H100 Qwen3 30B A3B short-run outcome without treating missing DeepEP/HybridEP packages as throughput data.
- Clarify that all-to-all is the correctness fallback when flex dispatcher packages are unavailable.

## Experiment
- Ran a short 16-GPU H100 Qwen3 30B A3B dispatcher-selection smoke on cw.
- All-to-all completed after disabling MoE permute fusion for the container-specific Triton JIT failure.
- DeepEP and HybridEP selected the requested flex backend but failed at model construction because their packages were not installed in the run container.

## Validation
- UV_CACHE_DIR=.local/uv-cache uv run pre-commit run --all-files
- No local unit tests run, per handoff.

### CODE REVIEW

**Comment by yaoyu-33** [comment]
Experiment handoff results for the short MoE dispatcher-selection run.

Context:
- Workload: Qwen3 30B A3B pretrain, H100 BF16, config variant v1
- Scale: 2 nodes, 8 GPUs/node, 16 GPUs total, EP=16
- Max steps: 5 for dispatcher cases
- Common overrides: `--moe_a2a_overlap false`, `--tokenizer_type NullTokenizer`, `comm_overlap.overlap_moe_expert_parallel_comm=false`, `comm_overlap.delay_wgrad_compute=false`, `model.moe_shared_expert_overlap=false`
- CUDA graph stack: recipe Transformer Engine graph defaults, scopes `moe_router,moe_preprocess`
- Local unit tests: not run, per handoff
- Pre-commit: `UV_CACHE_DIR=.local/uv-cache uv run pre-commit run --all-files` passed

Result table:

| Case | Backend arg | Effective dispatcher | Job | Status | Mean step after warmup | Median step after warmup | Mean TFLOP/s/GPU after warmup | Final loss | Peak max allocated | Train log |
|---|---|---|---|---|---:|---:|---:|---:|---:|---|
| `A_alltoall_nopermute` | `None` | `alltoall` | `11837549` | completed | 45.65 s | 44.28 s | 132.9 | 11.44050 | 61.351 GB | `/lustre/fsw/IOR-test/yuya/moe_dispatcher_20260517/logs/A_alltoall_nopermute_11837549_train.log` |
| `B_deepep_nopermute` | `deepep` | `flex` + `deepep` | `11837877` | failed before iter 1 | n/a | n/a | n/a | n/a | n/a | `/lustre/fsw/IOR-test/yuya/moe_dispatcher_20260517/logs/B_deepep_nopermute_11837877_train.log` |
| `C_hybridep_nopermute` | `hybridep` | `flex` + `hybridep` | `11837878` | failed before iter 1 | n/a | n/a | n/a | n/a | n/a | `/lustre/fsw/IOR-test/yuya/moe_dispatcher_20260517/logs/C_hybridep_nopermute_11837878_train.log` |

Additional artifacts:
- Slurm log, all-to-all: `/lustre/fsw/IOR-test/yuya/moe_dispatcher_20260517/logs/moe-disp_11837549.out`
- Slurm log, DeepEP: `/lustre/fsw/IOR-test/yuya/moe_dispatcher_20260517/logs/moe-disp_11837877.out`
- Slurm log, HybridEP: `/lustre/fsw/IOR-test/yuya/moe_dispatcher_20260517/logs/moe-disp_11837878.out`
- All-to-all config: `/lustre/fsw/IOR-test/yuya/moe_dispatcher_20260517/configs/A_alltoall_nopermute_11837549.yaml`
- DeepEP config: `/lustre/fsw/IOR-test/yuya/moe_dispatcher_20260517/configs/B_deepep_nopermute_11837877.yaml`
- HybridEP config: `/lustre/fsw/IOR-test/yuya/moe_dispatcher_20260517/configs/C_hybridep_nopermute_11837878.yaml`
- Local report: `.local/moe_dispatcher_selection_experiment_20260517.md`

All-to-all per-iteration data:

| Iteration | Step time | TFLOP/s/GPU | Loss |
|---:|---:|---:|---:|
| 1 | 87.8186 s | 68.7 | 12.34604 |
| 2 | 45.8394 s | 131.6 | 12.34616 |
| 3 | 42.5592 s | 141.7 | 12.12627 |
| 4 | 51.4850 s | 117.1 | 11.72187 |
| 5 | 42.7268 s | 141.2 | 11.44050 |

Caveats:
- This is a backend-availability and all-to-all fallback smoke, not a throughput comparison between all-to-all, DeepEP, and HybridEP.
- DeepEP and HybridEP selected the requested flex backend in dumped configs but failed before iteration 1 because the run container did not include the corresponding packages.
- The measured all-to-all path used `model.moe_permute_fusion=false`. A default all-to-all smoke, job `11837467`, reached the training loop but failed before iteration 1 in the fused MoE permutation Triton JIT path with `Unsupported function referenced: <function get_int_dtype ...>`.
- The run was intentionally short (`max_steps=5`) and should be rerun in a container with installed flex backends before making a production dispatcher decision.

**Review by cuichenx — APPROVED** [review]
